src/components/FunctionDefinition.js

```diff
@@ -4,10 +4,14 @@ import CodeDef from './CodeDef';
 import { graphql } from 'gatsby';
 
 const FunctionDefinition = ({ className, arguments: params, returnValue }) => {
-  if (!params.length && !returnValue.length) {
+  if (!params.length && !returnValue && !returnValue.length) {
```

Comment:
i think this line is evaluating differently than before, but im not sure if thats intended.

in particular, i think `!returnValue` will return false on empty array of return value. but i also see how now if passed something that isn't array, thats accounted for.

Reply:
Oh good catch!

I think we'll want to ensure the following:
* Are there no parameters AND:
  * Is the returnValue an array & is it empty OR:
    * Is it not an array and falsey

If that's correct, then here's what I've got:

```suggestion
  const hasParams = params.length;
  const isReturnArray = Array.isArray(returnValue);
  const hasReturn = (isReturnArray && returnValue.length) || (!isReturnArray && returnValue);
  
  if (!hasParams && !hasReturn) {
```

Does that seem right to you?

Reply:
i think this is right. would you happen to have any thoughts on splitting this into multiple if checks, each with a return for one condition.

Reply:
hmmm, let's see how that would look:

```suggestion
if (!params.length) return;
if (Array.isArray(returnValue) && !returnValue.length) return;
if (!Array.isArray(returnValue) && !returnValue) return;
```

That might be a little simpler (and could be combined with the boolean variables for maximum readability). Ideally we don't run into this situation that often - this code is mostly just extra defense against the SDK being in the wrong format.

Aaaanyway, what do you think?

Reply:
my thought was just around all the boolean logic in one line, and trying to think through it. i defer to you for best practice / experience 🙈  i would approve either.

Reply:
Alright - I'll probably go with the first one (one-line `if` statements make me nervous). Thanks for helping me work through that!